### ISSUE TITLE
clean up modelfinder output

### ISSUE THREAD

**Issue by roblanf** [iqtree/iqtree2#14]
Right now the modelfinder output is too verbose. See the e.g. of what it looks like below. In a terminal window this spreads over multiple lines, and makes the output useless.

I suggest we switch to something more like the partitionfinder output, which mostly looks like this:

```
INFO     | 2020-12-02 13:39:23,160 |       Finished subset 6077/14028, 43.32 percent done
INFO     | 2020-12-02 13:39:23,273 |       Finished subset 6078/14028, 43.33 percent done
INFO     | 2020-12-02 13:39:23,468 |       Finished subset 6079/14028, 43.33 percent done
INFO     | 2020-12-02 13:39:23,898 |       Finished subset 6080/14028, 43.34 percent done
INFO     | 2020-12-02 13:39:23,997 |       Finished subset 6081/14028, 43.35 percent done
INFO     | 2020-12-02 13:39:24,124 |       Finished subset 6082/14028, 43.36 percent done
INFO     | 2020-12-02 13:39:24,849 |       Finished subset 6083/14028, 43.36 percent done
INFO     | 2020-12-02 13:39:25,058 |       Finished subset 6084/14028, 43.37 percent done
INFO     | 2020-12-02 13:39:25,384 |       Finished subset 6085/14028, 43.38 percent done
INFO     | 2020-12-02 13:39:25,940 |       Finished subset 6086/14028, 43.38 percent done
INFO     | 2020-12-02 13:39:26,051 |       Finished subset 6087/14028, 43.39 percent done
INFO     | 2020-12-02 13:39:26,385 |       Finished subset 6088/14028, 43.40 percent done
INFO     | 2020-12-02 13:39:26,718 |       Finished subset 6089/14028, 43.41 percent done
INFO     | 2020-12-02 13:39:26,829 |       Finished subset 6090/14028, 43.41 percent done
INFO     | 2020-12-02 13:39:27,156 |       Finished subset 6091/14028, 43.42 percent done
INFO     | 2020-12-02 13:39:27,666 |       Finished subset 6092/14028, 43.43 percent done
INFO     | 2020-12-02 13:39:27,887 |       Finished subset 6093/14028, 43.43 percent done
INFO     | 2020-12-02 13:39:28,274 |       Finished subset 6094/14028, 43.44 percent done
INFO     | 2020-12-02 13:39:28,475 |       Finished subset 6095/14028, 43.45 percent done
INFO     | 2020-12-02 13:39:28,898 |       Finished subset 6096/14028, 43.46 percent done
```

and then only reports the merged subset that was chosen as the best at each step. 

Basically - we don't need to bombard the user with all the names of every merge we attempt. Instead, we need an output that will have a mostly fixed width (or at least, a width that changes rarely), so as it scrolls past at a million miles an hour users can see the progress.

```
19064 GTR+F+I+G4   560192.840  1.204       ES316890nucl_1stpos+ES319935nucl_1stpos+ES321806nucl_1stpos+ES321879nucl_1stpos+ES321882nucl_1stpos+O540nucl_1stpos+O6843nucl_1stpos+O8775nucl_1
stpos+O10295nucl_1stpos+O10419nucl_1stpos+O10567nucl_1stpos+O10821nucl_1stpos+O11092nucl_1stpos+O11094nucl_1stpos+O11722nucl_1stpos+O13507nucl_1stpos+O13660nucl_1stpos+O15665nucl_1stpos+O
16400nucl_1stpos+O17424nucl_1stpos+O19102nucl_1stpos+O19189nucl_1stpos+O19758nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O23857nucl_1stpos  0h:2m:52s (0h:1m:21s left)
19065 GTR+F+I+G4   560901.345  1.032       ES316508nucl_2ndpos+ES321882nucl_2ndpos+O988nucl_2ndpos+O988nucl_3rdpos+O6843nucl_2ndpos+O7569nucl_2ndpos+O8775nucl_2ndpos+O10419nucl_2ndpos+O10
567nucl_3rdpos+O10600nucl_1stpos+O10600nucl_2ndpos+O11635nucl_2ndpos+O11722nucl_2ndpos+O13507nucl_2ndpos+O17252nucl_2ndpos+O18599nucl_2ndpos+O19058nucl_2ndpos+O19102nucl_2ndpos+O21222nucl
_2ndpos+O21286nucl_2ndpos+O21944nucl_2ndpos+O23857nucl_2ndpos   0h:2m:53s (0h:1m:21s left)
19066 GTR+F+I+G4   560712.417  0.827       ES321879nucl_2ndpos+ES321882nucl_1stpos+O1187nucl_2ndpos+O5959nucl_2ndpos+O8818nucl_2ndpos+O10295nucl_1stpos+O10567nucl_1stpos+O10567nucl_2ndpos
+O10821nucl_1stpos+O10821nucl_2ndpos+O11092nucl_1stpos+O11094nucl_1stpos+O11094nucl_2ndpos+O11240nucl_2ndpos+O11569nucl_2ndpos+O11722nucl_1stpos+O13507nucl_1stpos+O13706nucl_2ndpos+O14581
nucl_2ndpos+O15665nucl_1stpos+O16052nucl_2ndpos+O16400nucl_1stpos+O16400nucl_2ndpos+O17424nucl_1stpos+O19102nucl_1stpos+O19189nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O22156nucl_2n
dpos+O22172nucl_2ndpos+O22441nucl_2ndpos+O23857nucl_1stpos      0h:2m:54s (0h:1m:22s left)
19067 GTR+F+I+G4   560435.179  0.698       ES316508nucl_2ndpos+ES321882nucl_2ndpos+O1187nucl_1stpos+O6843nucl_2ndpos+O7569nucl_2ndpos+O8775nucl_2ndpos+O8818nucl_1stpos+O10353nucl_1stpos+O
10353nucl_2ndpos+O10353nucl_3rdpos+O10419nucl_2ndpos+O11635nucl_1stpos+O11635nucl_2ndpos+O11722nucl_2ndpos+O12061nucl_2ndpos+O12201nucl_2ndpos+O13507nucl_2ndpos+O16052nucl_1stpos+O17252nu
cl_2ndpos+O18599nucl_2ndpos+O19102nucl_2ndpos+O21222nucl_1stpos+O21222nucl_2ndpos+O21286nucl_2ndpos+O21944nucl_2ndpos+O22172nucl_1stpos+O22441nucl_1stpos+O23857nucl_2ndpos     0h:2m:54s (
0h:1m:22s left)
19068 GTR+F+I+G4   560200.537  1.488       ES316508nucl_1stpos+O1187nucl_1stpos+O5959nucl_1stpos+O8818nucl_1stpos+O10353nucl_1stpos+O10353nucl_2ndpos+O10353nucl_3rdpos+O11240nucl_1stpos+O
11569nucl_1stpos+O11635nucl_1stpos+O12061nucl_1stpos+O12061nucl_2ndpos+O12201nucl_1stpos+O12201nucl_2ndpos+O13706nucl_1stpos+O14581nucl_1stpos+O16052nucl_1stpos+O17252nucl_1stpos+O18599nu
cl_1stpos+O19058nucl_1stpos+O21222nucl_1stpos+O21286nucl_1stpos+O22172nucl_1stpos+O22441nucl_1stpos     0h:2m:54s (0h:1m:22s left)
19069 GTR+F+I+G4   561382.102  2.125       ES321882nucl_1stpos+O9874nucl_3rdpos+O10295nucl_1stpos+O10567nucl_1stpos+O10821nucl_1stpos+O11092nucl_1stpos+O11094nucl_1stpos+O11722nucl_1stpos
+O13507nucl_1stpos+O15665nucl_1stpos+O16400nucl_1stpos+O17424nucl_1stpos+O17424nucl_3rdpos+O19102nucl_1stpos+O19189nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O22172nucl_3rdpos+O23857
nucl_1stpos+O30441nucl_3rdpos   0h:2m:55s (0h:1m:22s left)
19070 GTR+F+I+G4   560268.431  0.966       ES321882nucl_1stpos+O7569nucl_1stpos+O9874nucl_1stpos+O10295nucl_1stpos+O10567nucl_1stpos+O10821nucl_1stpos+O10867nucl_1stpos+O11092nucl_1stpos+
O11094nucl_1stpos+O11722nucl_1stpos+O13507nucl_1stpos+O14147nucl_1stpos+O15665nucl_1stpos+O16400nucl_1stpos+O17157nucl_1stpos+O17424nucl_1stpos+O17764nucl_1stpos+O19102nucl_1stpos+O19189n
ucl_1stpos+O21786nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O23857nucl_1stpos+O30441nucl_1stpos    0h:2m:55s (0h:1m:22s left)
19071 GTR+F+I+G4   560272.161  1.415       ES316508nucl_1stpos+ES321882nucl_1stpos+O5959nucl_1stpos+O10295nucl_1stpos+O10567nucl_1stpos+O10821nucl_1stpos+O11092nucl_1stpos+O11094nucl_1stp
os+O11240nucl_1stpos+O11569nucl_1stpos+O11722nucl_1stpos+O12061nucl_1stpos+O12201nucl_1stpos+O13507nucl_1stpos+O13706nucl_1stpos+O14581nucl_1stpos+O15665nucl_1stpos+O16400nucl_1stpos+O172
52nucl_1stpos+O17424nucl_1stpos+O18599nucl_1stpos+O19058nucl_1stpos+O19102nucl_1stpos+O19189nucl_1stpos+O21286nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O23857nucl_1stpos 0h:2m:55s (
0h:1m:22s left)
19072 GTR+F+I+G4   561142.489  1.197       ES321806nucl_2ndpos+O540nucl_2ndpos+O988nucl_2ndpos+O988nucl_3rdpos+O9874nucl_2ndpos+O10567nucl_3rdpos+O10600nucl_1stpos+O10600nucl_2ndpos+O1086
7nucl_2ndpos+O15665nucl_2ndpos+O17157nucl_2ndpos+O17764nucl_2ndpos+O19058nucl_2ndpos+O19758nucl_2ndpos+O21786nucl_2ndpos+O30441nucl_2ndpos      0h:2m:55s (0h:1m:22s left)
19073 GTR+F+I+G4   560396.428  9.169       ES316890nucl_3rdpos+O10295nucl_3rdpos+O11092nucl_3rdpos+O11569nucl_3rdpos+O11635nucl_3rdpos+O13507nucl_3rdpos+O13706nucl_3rdpos+O14581nucl_3rdpo
s+O16052nucl_3rdpos+O16400nucl_3rdpos+O17157nucl_3rdpos+O19189nucl_3rdpos+O19758nucl_3rdpos+O21786nucl_3rdpos+O21944nucl_3rdpos+O22156nucl_3rdpos+O22441nucl_3rdpos     0h:2m:55s (0h:1m:22
s left)
19074 GTR+F+I+G4   560986.855  3.010       ES316508nucl_1stpos+O5959nucl_1stpos+O9874nucl_3rdpos+O11240nucl_1stpos+O11569nucl_1stpos+O12061nucl_1stpos+O12201nucl_1stpos+O13706nucl_1stpos+
O14581nucl_1stpos+O17252nucl_1stpos+O17424nucl_3rdpos+O18599nucl_1stpos+O19058nucl_1stpos+O21286nucl_1stpos+O22172nucl_3rdpos+O30441nucl_3rdpos 0h:2m:56s (0h:1m:22s left)
19075 GTR+F+I+G4   560191.363  9.109       ES316890nucl_3rdpos+O9874nucl_3rdpos+O10295nucl_3rdpos+O11092nucl_3rdpos+O11569nucl_3rdpos+O11635nucl_3rdpos+O13507nucl_3rdpos+O13706nucl_3rdpos
+O14581nucl_3rdpos+O16400nucl_3rdpos+O17157nucl_3rdpos+O17424nucl_3rdpos+O21786nucl_3rdpos+O21944nucl_3rdpos+O22156nucl_3rdpos+O22172nucl_3rdpos+O22441nucl_3rdpos+O30441nucl_3rdpos    0h:
2m:56s (0h:1m:22s left)
19076 GTR+F+I+G4   561584.405  2.980       ES316890nucl_1stpos+ES319935nucl_1stpos+ES321806nucl_1stpos+ES321879nucl_1stpos+O540nucl_1stpos+O6843nucl_1stpos+O8775nucl_1stpos+O9874nucl_3rdp
os+O10419nucl_1stpos+O13660nucl_1stpos+O17424nucl_3rdpos+O19758nucl_1stpos+O22172nucl_3rdpos+O30441nucl_3rdpos  0h:2m:56s (0h:1m:22s left)
19077 GTR+F+I+G4   560158.457  0.385       ES316508nucl_2ndpos+ES316890nucl_2ndpos+ES319935nucl_2ndpos+ES321882nucl_2ndpos+O6843nucl_2ndpos+O7569nucl_2ndpos+O8775nucl_2ndpos+O10295nucl_2n
dpos+O10419nucl_2ndpos+O11092nucl_2ndpos+O11635nucl_2ndpos+O11722nucl_2ndpos+O13507nucl_2ndpos+O13660nucl_2ndpos+O14147nucl_2ndpos+O17252nucl_2ndpos+O17424nucl_2ndpos+O18599nucl_2ndpos+O1
9102nucl_2ndpos+O19189nucl_2ndpos+O21222nucl_2ndpos+O21286nucl_2ndpos+O21944nucl_2ndpos+O23857nucl_2ndpos       0h:2m:56s (0h:1m:23s left)
19078 GTR+F+I+G4   560403.485  1.676       ES316508nucl_1stpos+ES316890nucl_1stpos+ES319935nucl_1stpos+ES321806nucl_1stpos+ES321879nucl_1stpos+O540nucl_1stpos+O5959nucl_1stpos+O6843nucl_1
stpos+O8775nucl_1stpos+O10419nucl_1stpos+O11240nucl_1stpos+O11569nucl_1stpos+O12061nucl_1stpos+O12201nucl_1stpos+O13660nucl_1stpos+O13706nucl_1stpos+O14581nucl_1stpos+O17252nucl_1stpos+O1
8599nucl_1stpos+O19058nucl_1stpos+O19758nucl_1stpos+O21286nucl_1stpos   0h:2m:56s (0h:1m:23s left)
19079 GTR+F+I+G4   560153.284  1.061       ES321882nucl_1stpos+O1187nucl_1stpos+O8818nucl_1stpos+O10295nucl_1stpos+O10353nucl_1stpos+O10353nucl_2ndpos+O10353nucl_3rdpos+O10567nucl_1stpos+
O10821nucl_1stpos+O11092nucl_1stpos+O11094nucl_1stpos+O11635nucl_1stpos+O11722nucl_1stpos+O12061nucl_2ndpos+O12201nucl_2ndpos+O13507nucl_1stpos+O15665nucl_1stpos+O16052nucl_1stpos+O16400n
ucl_1stpos+O17424nucl_1stpos+O19102nucl_1stpos+O19189nucl_1stpos+O21222nucl_1stpos+O21944nucl_1stpos+O22156nucl_1stpos+O22172nucl_1stpos+O22441nucl_1stpos+O23857nucl_1stpos    0h:2m:56s (
0h:1m:23s left)
```

**Comment by roblanf** [comment]
Here's a suggested output format:

At the start of each step of an algorithm, you say what step and what algorithm it is and the current BIC score, e.g.

```
ModelFinder2    Greedy Algorithm    Step 1    125 Subsets    BIC 1091280.123
```

During the execution of each step, you overwrite a single line to make sure people know IQ-TREE is still working, something like

```
Finished subset 6088/14028    43.40 percent done    0h:2m:52s (0h:1m:21s left)
```

Then at the end of the process, we just start a new step of the algorithm. The only difference here is that after step 1 we can also report a delta BIC to let users know how things are going

```
ModelFinder2    Greedy Algorithm    Step 2    124 Subsets    BIC 1081280.123    deltaBIC -10000.000
```

So after a few steps of the algorithm the output on the screen (and in the log file) might look like:


```
ModelFinder2    Greedy Algorithm    Step 1    125 Subsets    BIC 1091280.123
ModelFinder2    Greedy Algorithm    Step 2    124 Subsets    BIC 1081280.123    deltaBIC -10000.000
ModelFinder2    Greedy Algorithm    Step 3    123 Subsets    BIC 1071280.123    deltaBIC -10000.000
ModelFinder2    Greedy Algorithm    Step 4    122 Subsets    BIC 1080280.123    deltaBIC -1000.000
ModelFinder2    Greedy Algorithm    Step 5    121 Subsets    BIC 1081180.123    deltaBIC -100.000
Finished subset 6088/14028    43.40 percent done    0h:2m:52s (0h:1m:21s left)
```

**Comment by roblanf** [comment]
@bqminh - turns out I already made a fairly detailed suggestion for what the output could look like.

What do you think? And how about you @thomaskf?

**Comment by thomaskf** [comment]
Hi @roblanf ,

Your suggestion about cleaning up the output for ModelFinder2 has been addressed in version 2.2.2.9 (https://github.com/iqtree/iqtree2/releases/tag/v2.2.2.9).

Thanks,
Thomas
